### PR TITLE
Refactor and remove CompositeAnswers to use JSON backend store

### PR DESCRIPTION
**Pull request by adammahmood**
### What is the context of this PR?
- Removes `CompositeAnswer` type and utilises `BasicAnswer` type for `DateRange` answers

### How to review 
- Ensure you can add/modify/delete `DateRange` answers
- Ensure you can add/modify/delete `DateRange` validations

### CODE REVIEW

**Comment by adammahmood** [comment]
[CompositeAnswers](https://trello.com/c/Q8oGjKx2/841-compositeanswers)

File: eq-author-api/schema/resolvers/base.js [hunk 1]
```diff
@@ -907,22 +905,12 @@ const Resolvers = {
       return parentPage;
     },
     validation: answer =>
-      ["number", "date"].includes(getValidationEntity(answer.type))
+      ["number", "date", "dateRange"].includes(getValidationEntity(answer.type))
```
Comment (tgandrews) [inline eq-author-api/schema/resolvers/base.js]: We should probably make these constants

File: eq-author-api/schema/typeDefs.js [hunk 2]
```diff
@@ -138,27 +139,14 @@ type MultipleChoiceAnswer implements Answer {
...
   label: String
+  secondaryLabel: String
   type: AnswerType!
   options: [Option]
   mutuallyExclusiveOption: Option
   page: QuestionPage
   properties: JSON
 }
 
-type CompositeAnswer implements Answer {
```
Comment (tgandrews) [inline eq-author-api/schema/typeDefs.js]: If this doesn't exist can we get rid of `BasicAnswer` and just have `Answer`?

Comment (adammahmood) [reply eq-author-api/schema/typeDefs.js]: Attempting to keep changes to a minimum for the moment. Also I'm pretty sure we would get a naming conflict with `Answer` interface.

**Comment by adammahmood** [comment]
@tgandrews Linting has a few failures.  We'll pick this up when we do a 2nd pass on everything to tidy it up.  I'm going to tackle all of the publishing changes as part of https://trello.com/b/GOqjEepx/eq-author

File: eq-author-api/schema/typeDefs.js (hunk 2 above)
Comment (tgandrews) [reply eq-author-api/schema/typeDefs.js]: Can we just remove that interface?

Comment (adammahmood) [reply eq-author-api/schema/typeDefs.js]: It results in a lot of changes on the UI (which we're trying to leave as untouched as possible).

File: eq-author-api/schema/resolvers/base.js (hunk 1 above)
Comment (adammahmood) [reply eq-author-api/schema/resolvers/base.js]: This references values defined here: https://github.com/ONSdigital/eq-author-app/blob/39144650e141d176b0e66b37ab15c0cc0eb314ea/eq-author-api/utils/defaultAnswerValidations.js#L15 

Should we still convert to constants?

**Pull request merged**